geo/src/algorithm/outlier_detection.rs

```diff
@@ -0,0 +1,446 @@
+    /// over the data using a range of `k_neighbours` values, then aggregating these, taking e.g.
+    /// the minimum or the mean.
+    ///
+    /// # Note on Erroneous Input
+    /// If `k_neighbours` >= points in the set, or `k_neighbours` < 1, all input points will be returned with an LOF score of 1.
+    /// If there are at least `k_neighbours` duplicate points of an input point, LOF scores can be `∞` or `NaN`.
+    /// It is thus advisable to **deduplicate** or otherwise ensure the uniqueness of the input points.
+    ///
+    /// # Note on Returned Points
+    /// Outlier scores are always returned corresponding to input point order
+    ///
+    /// # Examples
+    ///
+    /// ## MultiPoint
+    ///
+    /// ```
+    /// use approx::assert_relative_eq;
+    /// use geo::OutlierDetection;
+    /// use geo::{point, MultiPoint};
+    ///
+    /// let v = vec![
+    ///     point!(x: 0.0, y: 0.0),
+    ///     point!(x: 0.0, y: 1.0),
+    ///     point!(x: 3.0, y: 0.0),
+    ///     point!(x: 1.0, y: 1.0),
+    /// ];
+    ///
+    /// let lofscores = v.outliers(2);
+    /// // The third point is an outlier, resulting in a large LOF score
+    /// assert_relative_eq!(lofscores[2], 3.0);
+    /// // The last point is an inlier, resulting in a small LOF score
+    /// assert_relative_eq!(lofscores[3], 1.0);
+    /// ```
+    ///
+    /// ## Computing indices, sorting by LOF score
+    ///```
+    /// use geo::OutlierDetection;
+    /// use geo::{point, MultiPoint};
+    ///
+    /// // these points contain 4 strong outliers
+    /// let v = vec![
+    ///     point!(x: 0.16, y: 0.14),
+    ///     point!(x: 0.15, y: 0.33),
+    ///     point!(x: 0.37, y: 0.25),
+    ///     point!(x: 0.3 , y: 0.4),
+    ///     point!(x: 0.3 , y: 0.1),
+    ///     point!(x: 0.3 , y: 0.2),
+    ///     point!(x: 1.3 , y: 2.3),
+    ///     point!(x: 1.7 , y: 0.2),
+    ///     point!(x: 0.7 , y: -0.9),
+    ///     point!(x: 0.21, y: 2.45),
+    ///     point!(x: 0.8 , y: 0.7),
+    ///     point!(x: 0.9 , y: 0.7),
+    ///     point!(x: 0.8 , y: 0.6),
+    ///     point!(x: 0.73, y: 0.65),
+    ///     point!(x: 0.9 , y: 0.6),
+    ///     point!(x: 1.0, y: 0.6),
+    ///     point!(x: 1.0, y: 0.7),
+    ///     point!(x: 0.25, y: 0.29),
+    ///     point!(x: 0.2 , y: 0.2),
+    /// ];
+    /// let lofs = &mut v.outliers(3);
+    /// let mut idx_lofs: Vec<(usize, f64)> = lofs
+    ///     .iter()
+    ///     .enumerate()
+    ///     .map(|(idx, score)| (idx, *score))
+    ///     .collect();
+    /// // sort by LOF score
+    /// idx_lofs.sort_by(|a, b| a.1.partial_cmp(&b.1).unwrap());
+    /// // most likely outliers first
+    /// idx_lofs.reverse();
+    /// // four outliers, LOF scores way above 10
+    /// idx_lofs
+    ///     .iter()
+    ///     .take(4)
+    ///     .for_each(|score| assert!(score.1 > 10.0));
+    ///```
+    fn outliers(&self, k_neighbours: usize) -> Vec<T>;
```

Comment:
You changed this to `outlier_scores` in response to my feedback, and then changed it back to `outliers`. I just wanted to make sure that was intentional.

Reply:
Yep! Sorry.